### ISSUE TITLE
Input field autofill popup

### ISSUE THREAD

**Issue by suntong** [maxence-charriere/go-app#863]
For `app.Input().Type("text")`, what's the trick to allow browser autofill so that people can select old values from the popup? 

(The code I copied from demo have it, but its input component that I copied to my own project doesn't have the autofill, but the code looks identical to me)

Please help. Thanks.

**Comment by mlctrez** [comment]
This seems to work pretty well, pulled from this stackoverflow answer:
https://stackoverflow.com/questions/7223168/how-to-trigger-autofill-in-google-chrome/41965106#41965106

```go

app.Form().
	Body(
		app.Label().For("frmNameA").Body(app.Text("Name")),
		app.Input().Type("text").Name("name").ID("frmNameA").Placeholder("Full name").
			Required(true).Attr("autocomplete", "name"),
		app.Label().For("frmEmailA").Body(app.Text("Email")),
		app.Input().Type("email").Name("email").ID("frmEmailA").Placeholder("name@example.com").
			Required(true).Attr("autocomplete", "email"),
		app.Label().For("frmEmailC").Body(app.Text("Confirm Email")),
		app.Input().Type("email").Name("emailC").ID("frmEmailC").Placeholder("name@example.com").
			Required(true).Attr("autocomplete", "email"),
		app.Label().For("frmPhoneNumA").Body(app.Text("Phone")),
		app.Input().Type("tel").Name("phone").ID("frmPhoneNumA").Placeholder("+1-555-555-1212").
			Required(true).Attr("autocomplete", "tel"),
	)
```

**Comment by mlctrez** [comment]
Also, raising visibility to https://pojntfx.github.io/html2goapp/ again.  

This can be used to take arbitrary html and produce go-app components.  It is not perfect but can save some typing!

**Comment by suntong** [comment]
Thanks for the answer.

> (The code I copied from demo have it, but its input component that I copied to my own project doesn't have the autofill, but the code looks identical to me)

Turned out it is the _field name_ that make the difference! As I have a special field that I want to have autocomplete values of its own, i.e. not to use the common `name`, `given-name`, etc to pollute each other. But it is not working, despite filling the form many times. However, right after I change the autocomplete attribute back to e.g. `name`, `given-name`, etc it's working again.

I've documented my testings at:

https://github.com/suntong/go-app-demos/blob/eaabea9380ca9d675607983202e1200b345a8f82/0A2A-hello/main.go#L40-L43

According to https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#autofill-detail-tokens, all of my above test cases should be working.

Does anyone has any idea please?

**Comment by mlctrez** [comment]
Some of the credit card fields will require a secure connection to work correctly.

Looking at https://www.w3.org/WAI/GL/wiki/WCAG_2.1_Implementations/JF/research#Functional_Gap_in_Browsers_with_.40autocomplete_Attributes 
there are quite a few gaps in what is actually supported versus what is listed as possible in the whatwg page.  I also realize this testing is out of date by a few years so that may have changed.

I would suggest getting the form complete working in a standard html page and then trying to see what is not supported in go-app.  Generally if you can build it in plain html it will end up working in go-app.

**Comment by oderwat** [comment]
> I'm not using **form submit**

Why not? You can call it and just do nothing.

**Comment by suntong** [comment]
> Why not? You can call it and just do nothing.

OK, I'll look it up and see how to do it. 

> there are quite a few gaps in what is actually supported versus what is listed

Wow, thanks a lot for that url, which I would never be able to find it myself. 

Telling from the limited supported list, I found that my special field & true fields cannot avoid polluting each other, so I decided give up using autocomplete, and try supply my own drop down list instead (using the datalist element: developer.mozilla.org/en-US/docs/Web/HTML/Element/datalist).

Thank you all for the help that I'd never be able to find elsewhere ❤️❤️
